functions/Remove-DbaDbTableData.ps1

```diff
@@ -0,0 +1,306 @@
+function Remove-DbaDbTableData {
+    <#
+    .SYNOPSIS
+        Removes table data using a batch technique from a database(s) for each instance(s) of SQL Server.
+
+    .DESCRIPTION
+        This command does a batch delete of table data using the technique described by Aaron Bertrand here: https://sqlperformance.com/2013/03/io-subsystem/chunk-deletes. The main goal of this command is to ensure that the log file size is controlled while deleting data. This command can be used for doing both very large deletes or small deletes. Foreign keys are not temporarily removed, so the caller needs to perform deletes in the correct order with dependent tables or enable cascading deletes. When a database is using the full or bulk_logged recovery model this command will take log backups at the end of each batch. If the database is using the simple recovery model then CHECKPOINTs will be performed. The object returned will contain metadata about the batch deletion process including the log backup details.
+
+    .PARAMETER SqlInstance
```

Comment:
We likely need some validation control for this parameter to ensure it is not an Azure SQL instance because this command won't work against those. Since it is not calling `Connect-SqlInstance` directly we can't use the `-AzureUnsupported`

Reply:
@lancasteradam : is this still valid ? I feel that removing data in batches when done on Azure is just a matter of skipping the log backup no matter what, given backups are "managed"

Reply:
@niphlod, I think it is still valid. I like the suggestion that @wsmelton made about ensuring it is not an Azure SQL instance. I can add validation code for that. This will keep the command simpler and focused for on-prem scenarios. Let me know if you agree with that approach.

Reply:
hum, I'm not convinced...why can't we allow removing table data in batches for Azure ? It's Remove-DbaDbTableData, not Remove-DbaDbTableDataAndBackup. @wsmelton , @Stuart-Moore  your thoughts on this ? I feel we can make so that if you don't pass a LogBackupPath backups are not taken (and issue a warning?). This could "solve" the issue of "shoving a log backup".
IMHO it's perfectly fine to "defaulting" to on-premise "logic" of "remove-and-backup" but on Azure you can just leverage the "remove" logic without the backup part. BTW @lancasteradam sorry this is taking so long but it's going to be an outstanding function ^_^

Reply:
The cautionary tale that exist with Azure SQL is that because you don't control the backups large batch deletion can cause 40552 errors. This means Azure SQL killed the transaction because of log space.

If error handling is in place to catch that properly, then I'm good with including it. Users will just have to be aware of the limits with deleting data on Azure SQL DBs.

Reply:
that try/catch should intercept it and later on the function checks for an empty errormessage before continuing so I think it's perfectly managed there. At this point the missing bit to accomodate all reviews is to make the backup optional (if a LogBackupPath is not passed).